Sources/Load.php

```diff
@@ -407,7 +407,7 @@ function loadUserSettings()
 				$check = true;
 			// SHA-1 passwords should be 40 characters long.
 			elseif (strlen($password) == 40)
-				$check = sha1($user_settings['passwd'] . $user_settings['password_salt']) == $password;
+				$check = hash_hmac('sha1', sha1($user_settings['passwd'] . $user_settings['password_salt']), get_auth_secret()) == $password;
```

Comment:
We might need to test whether authentication still works as intended with our changes. I'm not immediately sure what the "auth secret" is, but I have a bad feeling about this.